# Borrar columnas

Usando dataset 02desastres_paralimpiar.csv

Objetivos:

Crear dataset 02desastres_colsreducidas.csv en ../Data/02ParaLimpiar/ sin columnas innecesarias

## Importar librerías

In [34]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Cargar datos

In [35]:
df = pd.read_csv('../../Data/02ParaLimpiar/02desastres_paralimpiar.csv', encoding='latin-1', delimiter=';', engine='python')
# encoding='utf8' da problemas
# dejar engine='python' para evitar advertencias

# Estadísticas básicas (columnas originales)

### Muestras head

In [36]:
df.head(5)

Dis No  Year   Seq Glide Disaster Group Disaster Subgroup   
0  1900-9002-CPV  1900  9002   NaN        Natural    Climatological  \
1  1900-9001-IND  1900  9001   NaN        Natural    Climatological   
2  1902-0012-GTM  1902    12   NaN        Natural       Geophysical   
3  1902-0003-GTM  1902     3   NaN        Natural       Geophysical   
4  1902-0010-GTM  1902    10   NaN        Natural       Geophysical   

       Disaster Type Disaster Subtype Disaster Subsubtype   Event Name  ...   
0            Drought          Drought                 NaN          NaN  ...  \
1            Drought          Drought                 NaN          NaN  ...   
2         Earthquake  Ground movement                 NaN          NaN  ...   
3  Volcanic activity         Ash fall                 NaN  Santa Maria  ...   
4  Volcanic activity         Ash fall                 NaN  Santa Maria  ...   

  Reconstruction Costs Adj Insured Damages Insured Damages Adj Total Damages   
0                      NaN             NaN                 NaN           NaN  \
1                      NaN             NaN                 NaN           NaN   
2                      NaN             NaN                 NaN       25000.0   
3                      NaN             NaN                 NaN           NaN   
4                      NaN             NaN                 NaN           NaN   

  Total Damages Adj          CPI Adm Level Admin1 Code Admin2 Code   
0               NaN  2,849084409       NaN         NaN         NaN  \
1               NaN  2,849084409       NaN         NaN         NaN   
2          843726.0  2,963047785       NaN         NaN         NaN   
3               NaN  2,963047785       NaN         NaN         NaN   
4               NaN  2,963047785       NaN         NaN         NaN   

  Geo Locations  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  

[5 rows x 50 columns]

## shape, info, dtypes

In [37]:
df.shape
# Resultado: 16636 filas y 50 columnas)

(16636, 50)

In [38]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16636 entries, 0 to 16635
Data columns (total 50 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Dis No                    16636 non-null  object 
 1   Year                      16636 non-null  int64  
 2   Seq                       16636 non-null  int64  
 3   Glide                     1736 non-null   object 
 4   Disaster Group            16636 non-null  object 
 5   Disaster Subgroup         16636 non-null  object 
 6   Disaster Type             16636 non-null  object 
 7   Disaster Subtype          13313 non-null  object 
 8   Disaster Subsubtype       1117 non-null   object 
 9   Event Name                3969 non-null   object 
 10  Country                   16636 non-null  object 
 11  ISO                       16636 non-null  object 
 12  Region                    16636 non-null  object 
 13  Continent                 16636 non-null  object 
 14  Locati

In [39]:
df.dtypes

Dis No                       object
Year                          int64
Seq                           int64
Glide                        object
Disaster Group               object
Disaster Subgroup            object
Disaster Type                object
Disaster Subtype             object
Disaster Subsubtype          object
Event Name                   object
Country                      object
ISO                          object
Region                       object
Continent                    object
Location                     object
Origin                       object
Associated Dis               object
Associated Dis2              object
OFDA Response                object
Appeal                       object
Declaration                  object
AID Contribution            float64
Dis Mag Value               float64
Dis Mag Scale                object
Latitude                     object
Longitude                    object
Local Time                   object
River Basin                 

## Columnas

In [40]:
df.columns

Index(['Dis No', 'Year', 'Seq', 'Glide', 'Disaster Group', 'Disaster Subgroup',
       'Disaster Type', 'Disaster Subtype', 'Disaster Subsubtype',
       'Event Name', 'Country', 'ISO', 'Region', 'Continent', 'Location',
       'Origin', 'Associated Dis', 'Associated Dis2', 'OFDA Response',
       'Appeal', 'Declaration', 'AID Contribution', 'Dis Mag Value',
       'Dis Mag Scale', 'Latitude', 'Longitude', 'Local Time', 'River Basin',
       'Start Year', 'Start Month', 'Start Day', 'End Year', 'End Month',
       'End Day', 'Total Deaths', 'No Injured', 'No Affected', 'No Homeless',
       'Total Affected', 'Reconstruction Costs', 'Reconstruction Costs Adj',
       'Insured Damages', 'Insured Damages Adj', 'Total Damages',
       'Total Damages Adj', 'CPI', 'Adm Level', 'Admin1 Code', 'Admin2 Code',
       'Geo Locations'],
      dtype='object')

### Columnas con valores numéricos: 21

Year                                           int64, 
Seq                                            int64, 
AID Contribution                             float64, 
Dis Mag Value                                float64, 
Year                                         int64, 
Start Month                                  float64, 
Start Day                                    float64, 
End Year                                       int64, 
End Month                                    float64, 
End Day                                      float64, 
Total Deaths                                 float64, 
No Injured                                   float64, 
No Affected                                  float64, 
No Homeless                                  float64, 
Total Affected                               float64, 
Reconstruction Costs                         float64, 
Reconstruction Costs, Adjusted               float64, 
Insured Damages                              float64, 
Insured Damages, Adjusted                    float64, 
Total Damages                                float64, 
Total Damages, Adjusted                      float64

### Columnas con variables categóricas: 25

Dis No, Glide, Disaster Group, Disaster Subgroup, Disaster Type, Disaster Subtype, Disaster Subsubtype, Event Name, Country, ISO, Region, Continent, Location, Origin, Associated Dis, Associated Dis2, OFDA Response, Appeal, Declaration, Dis Mag Scale, Latitude, Longitude, Local Time, River Basin

## Categorización de variables numéricas y categóricas por niveles según clasificación de dataset

In [41]:
num_lvl1 = ['Year', 'Seq']
categ_lvl1_id = ['Dis No']
categ_lvl1 = ['Disaster Subgroup', 'Disaster Type','Disaster Subtype']
categ_lvl2_geo = ['Country', 'ISO', 'Region', 'Continent', 'Location'] # 'River Basin'
categ_lvl2_geoloc = ['Latitude', 'Longitude'] # Estas pertenecen a la anterior pero es mejor separarlas debido a su carácter más específico
num_lvl2_temp = ['Start Year','Start Month', 'End Year', 'End Month'] # Son objects, evaluar si es necesario convertir a datetimes numéricas
# De num_lvl2_temp eliminar: 'Local Time', 'Start Day', 'End Day'
# categ_lvl2_time = ['Local Time']
categ_lvl2_fis = ['Origin', 'Associated Dis','Associated Dis2', 'Dis Mag Scale']
categ_lvl2_status = ['AID Contribution', 'OFDA Response', 'Appeal','Declaration'] # Coincidir con World Risk Index
# categ_lvl3_orgs = [] Se borraron los Adm Level y Admin Codes
# num_lvl3_human_impact = ['Total Deaths', 'No Injured', 'No Affected', 'No Homeless', 'Total Affected']
# num_lvl3_economic_impact = ['Reconstruction Costs', 'Reconstruction Costs Adj', 'Insured Damages', 'Insured Damages Adj', 'Total Damages', 'Total Damages Adj']

## Criterio de borrado de columnas:
* Demasiados nulls
* Categoría de columnas se puede encontrar más completa en dataset complementario
* Reducir dataset para acotar proyecto

## Valores faltantes

In [42]:
df.isnull().sum()

Dis No                          0
Year                            0
Seq                             0
Glide                       14900
Disaster Group                  0
Disaster Subgroup               0
Disaster Type                   0
Disaster Subtype             3323
Disaster Subsubtype         15519
Event Name                  12667
Country                         0
ISO                             0
Region                          0
Continent                       0
Location                     1811
Origin                      12551
Associated Dis              13043
Associated Dis2             15873
OFDA Response               14920
Appeal                      14077
Declaration                 13293
AID Contribution            15860
Dis Mag Value               11572
Dis Mag Scale                1220
Latitude                    13861
Longitude                   13861
Local Time                  15480
River Basin                 15300
Start Year                      0
Start Month   

In [43]:
df.isnull()

Dis No   Year    Seq  Glide  Disaster Group  Disaster Subgroup   
0       False  False  False   True           False              False  \
1       False  False  False   True           False              False   
2       False  False  False   True           False              False   
3       False  False  False   True           False              False   
4       False  False  False   True           False              False   
...       ...    ...    ...    ...             ...                ...   
16631   False  False  False   True           False              False   
16632   False  False  False  False           False              False   
16633   False  False  False   True           False              False   
16634   False  False  False   True           False              False   
16635   False  False  False  False           False              False   

       Disaster Type  Disaster Subtype  Disaster Subsubtype  Event Name  ...   
0              False             False                 True        True  ...  \
1              False             False                 True        True  ...   
2              False             False                 True        True  ...   
3              False             False                 True       False  ...   
4              False             False                 True       False  ...   
...              ...               ...                  ...         ...  ...   
16631          False             False                 True        True  ...   
16632          False             False                 True       False  ...   
16633          False             False                 True       False  ...   
16634          False             False                 True        True  ...   
16635          False             False                 True       False  ...   

       Reconstruction Costs Adj  Insured Damages  Insured Damages Adj   
0                          True             True                 True  \
1                          True             True                 True   
2                          True             True                 True   
3                          True             True                 True   
4                          True             True                 True   
...                         ...              ...                  ...   
16631                      True             True                 True   
16632                      True             True                 True   
16633                      True             True                 True   
16634                      True             True                 True   
16635                      True             True                 True   

       Total Damages  Total Damages Adj    CPI  Adm Level  Admin1 Code   
0               True               True  False       True         True  \
1               True               True  False       True         True   
2              False              False  False       True         True   
3               True               True  False       True         True   
4               True               True  False       True         True   
...              ...                ...    ...        ...          ...   
16631           True               True   True      False        False   
16632           True               True   True      False        False   
16633           True               True   True       True         True   
16634           True               True   True       True         True   
16635           True               True   True      False        False   

       Admin2 Code  Geo Locations  
0             True           True  
1             True           True  
2             True           True  
3             True           True  
4             True           True  
...            ...            ...  
16631        False          False  
16632         True          False  
16633         True           True  
16634         True           True  
16635  

### Columnas (numéricas y categóricas) con más valores faltantes

Numéricas con más nulls:
AID Contribution            15860
Dis Mag Value               11572
No Injured                  12489
No Homeless                 14166
Reconstruction Costs        16598
Reconstruction Costs Adj    16600
Insured Damages             15527
Insured Damages Adj         15527
Total Damages               11252
Total Damages Adj           11270

Categóricas con más nulls:
Origin                      12551
Associated Dis              13043
Associated Dis2             15873
OFDA Response               14920
Appeal                      14077
Declaration                 13293
Latitude                    13861
Longitude                   13861
Local Time                  15480
River Basin                 15300

## Borrar columnas

In [44]:
df.drop(columns=['Event Name', 'Disaster Group', 'Disaster Subsubtype', 'Glide', 'Local Time', 'Start Day', 'End Day', 'River Basin','Total Deaths', 'No Injured', 'No Affected', 'No Homeless', 'Total Affected', 'Reconstruction Costs', 'Reconstruction Costs Adj', 'Insured Damages', 'Insured Damages Adj', 'Total Damages', 'Total Damages Adj', 'Adm Level', 'Admin1 Code', 'Admin2 Code', 'Geo Locations'], inplace=True)

## Estadísticas básicas posteriores a borrar columnas

In [45]:
df.columns

Index(['Dis No', 'Year', 'Seq', 'Disaster Subgroup', 'Disaster Type',
       'Disaster Subtype', 'Country', 'ISO', 'Region', 'Continent', 'Location',
       'Origin', 'Associated Dis', 'Associated Dis2', 'OFDA Response',
       'Appeal', 'Declaration', 'AID Contribution', 'Dis Mag Value',
       'Dis Mag Scale', 'Latitude', 'Longitude', 'Start Year', 'Start Month',
       'End Year', 'End Month', 'CPI'],
      dtype='object')

In [46]:
df.shape

(16636, 27)

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16636 entries, 0 to 16635
Data columns (total 27 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Dis No             16636 non-null  object 
 1   Year               16636 non-null  int64  
 2   Seq                16636 non-null  int64  
 3   Disaster Subgroup  16636 non-null  object 
 4   Disaster Type      16636 non-null  object 
 5   Disaster Subtype   13313 non-null  object 
 6   Country            16636 non-null  object 
 7   ISO                16636 non-null  object 
 8   Region             16636 non-null  object 
 9   Continent          16636 non-null  object 
 10  Location           14825 non-null  object 
 11  Origin             4085 non-null   object 
 12  Associated Dis     3593 non-null   object 
 13  Associated Dis2    763 non-null    object 
 14  OFDA Response      1716 non-null   object 
 15  Appeal             2559 non-null   object 
 16  Declaration        334

In [48]:
df.head()

Dis No  Year   Seq Disaster Subgroup      Disaster Type   
0  1900-9002-CPV  1900  9002    Climatological            Drought  \
1  1900-9001-IND  1900  9001    Climatological            Drought   
2  1902-0012-GTM  1902    12       Geophysical         Earthquake   
3  1902-0003-GTM  1902     3       Geophysical  Volcanic activity   
4  1902-0010-GTM  1902    10       Geophysical  Volcanic activity   

  Disaster Subtype     Country  ISO           Region Continent  ...   
0          Drought  Cabo Verde  CPV   Western Africa    Africa  ...  \
1          Drought       India  IND    Southern Asia      Asia  ...   
2  Ground movement   Guatemala  GTM  Central America  Americas  ...   
3         Ash fall   Guatemala  GTM  Central America  Americas  ...   
4         Ash fall   Guatemala  GTM  Central America  Americas  ...   

  AID Contribution Dis Mag Value Dis Mag Scale Latitude Longitude Start Year   
0              NaN           NaN           Km2      NaN       NaN       1900  \
1              NaN           NaN           Km2      NaN       NaN       1900   
2              NaN           8.0       Richter       14       -91       1902   
3              NaN           NaN           NaN      NaN       NaN       1902   
4              NaN           NaN           NaN      NaN       NaN       1902   

  Start Month  End Year  End Month          CPI  
0         NaN      1900        NaN  2,849084409  
1         NaN      1900        NaN  2,849084409  
2         4.0      1902        4.0  2,963047785  
3         4.0      1902        4.0  2,963047785  
4        10.0      1902       10.0  2,963047785  

[5 rows x 27 columns]

# Guardar dataset en csv

In [49]:
df.to_csv('../../Data/02ParaLimpiar/02desastres_colsreducidas.csv', index=False)

Dataset con reducción de columnas. Queda limpiar nulls